datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -680,6 +680,7 @@ pub fn like_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataTyp
     string_coercion(lhs_type, rhs_type)
         .or_else(|| dictionary_coercion(lhs_type, rhs_type, false))
         .or_else(|| null_coercion(lhs_type, rhs_type))
+        .or_else(|| binary_coercion(lhs_type, rhs_type))
```

Comment:
This coerces string to binary, but currently Like expr only works on string array. I saw you did a commit which adds binary support to Like but it was removed.